tools/infer/text/utils/recovery_to_doc.py

```diff
@@ -0,0 +1,153 @@
+import os
+from typing import Dict, List
+
+from docx import Document, shared
+from docx.enum.section import WD_SECTION
+from docx.enum.text import WD_ALIGN_PARAGRAPH
+from docx.oxml.ns import qn
+
+from .table_process import HtmlToDocx
+
+
+def set_document_styles(doc: Document) -> None:
+    """
+    Set the styles for the document.
+    Args:
+        doc (Document): The document to set styles for.
+    """
+    doc.styles["Normal"].font.name = "Times New Roman"
+    doc.styles["Normal"]._element.rPr.rFonts.set(qn("w:eastAsia"), "宋体")
+    doc.styles["Normal"].font.size = shared.Pt(6.5)
+
+
+def convert_info_docx(res: List[Dict], save_folder: str, doc_name: str) -> None:
+    """
+    Convert OCR results to a DOCX file.
+    Args:
+        res (List[Dict]): OCR results.
+        save_folder (str): Folder to save the DOCX file.
+        doc_name (str): Name of the DOCX file.
+    Returns:
+        None
+    """
+    doc = Document()
+    set_document_styles(doc)
+
+    flag = 1  # Current layout flag
+    previous_layout = None  # To record the previous layout
+
+    for region in res:
+        if not region["res"]:
+            continue
+
+        # Check if the current layout has changed to avoid creating the same layout repeatedly
+        current_layout = region["layout"]
+        if current_layout != previous_layout:
+            section = doc.add_section(WD_SECTION.CONTINUOUS)
+            if current_layout == "single":
+                section._sectPr.xpath("./w:cols")[0].set(qn("w:num"), "1")
+                flag = 1
+            elif current_layout == "double":
+                section._sectPr.xpath("./w:cols")[0].set(qn("w:num"), "2")
+                flag = 2
+            elif current_layout == "triple":
+                section._sectPr.xpath("./w:cols")[0].set(qn("w:num"), "3")
+                flag = 3
+            previous_layout = current_layout  # Update the previous layout record
+
+        # Insert content based on the region type
+        if region["type"].lower() == "figure":
+            img_path = region["res"]
+            paragraph_pic = doc.add_paragraph()
+            paragraph_pic.alignment = WD_ALIGN_PARAGRAPH.CENTER
+            run = paragraph_pic.add_run("")
+            # Insert picture, width depends on the column layout
+            if flag == 1:
+                run.add_picture(img_path, width=shared.Inches(5))
+            elif flag == 2:
+                run.add_picture(img_path, width=shared.Inches(2.5))
+            elif flag == 3:
+                run.add_picture(img_path, width=shared.Inches(1.5))
+
+        elif region["type"].lower() == "title":
+            doc.add_heading(region["res"])
+
+        elif region["type"].lower() == "table":
+            parser = HtmlToDocx()
+            parser.table_style = "TableGrid"
+            parser.handle_table(region["res"], doc)
+
+        else:  # Default to handling text regions
```

Comment:
ibid, Comments can be written in a separate line.

Reply:
done